rosidl_generator_py/rosidl_generator_py/generate_py_impl.py

```diff
@@ -52,15 +56,18 @@ def generate_py(generator_arguments_file, typesupport_impls):
         'get_python_type': get_python_type,
         'primitive_msg_type_to_c': primitive_msg_type_to_c,
         'value_to_py': value_to_py,
+        'convert_camel_case_to_lower_case_underscore': convert_camel_case_to_lower_case_underscore,
     }
     latest_target_timestamp = get_newest_modification_time(args['target_dependencies'])
 
     modules = defaultdict(list)
+    message_specs = []
     for ros_interface_file in args['ros_interface_files']:
         extension = os.path.splitext(ros_interface_file)[1]
         subfolder = os.path.basename(os.path.dirname(ros_interface_file))
-        if extension == '.msg':
+        if extension == '.msg' and subfolder == 'msg':
```

Comment:
What happens to the request and response messages of services if they are being excluded here?